### PR TITLE
build system: add VERBOSE_ASSERT flag

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Add a compile-time convenience variable to enable verbose asserts.

### Issues/PRs references

alternative to #16864

### CODE REVIEW

File: Makefile.include [hunk 1]
```diff
@@ -284,6 +284,16 @@ else
...
 endif
 
+# Set this to 1 to enable pretty-printing the assert location in addition
+# to the address of the failing assert.
+# This saves you to resolve the address manually with addr2line at the cost
+# of greater ROM consumption.
+VERBOSE_ASSERT ?= 0
+ifeq ($(VERBOSE_ASSERT),1)
+  CFLAGS += -DDEBUG_ASSERT_VERBOSE
+  DEVELHELP = 1
```
Comment (maribu) [inline Makefile.include:294]: How about `:=` instead of `=`?

**Review by maribu — APPROVED** [review]
lgtm

**Comment by miri64** [comment]
Can I haz documentation in `makfiles/vars.inc.mk`, plz? ;-)

**Comment by maribu** [comment]
> Can I haz documentation in `makfiles/vars.inc.mk`, plz? ;-)

Done: https://github.com/RIOT-OS/RIOT/pull/16894

**Pull request merged**